### PR TITLE
ingress-egress-tracker: add tx_ref to redis

### PR DESCRIPTION
**Pull request by marcellorigotti**
# Pull Request

Closes: PRO-1196

## Checklist

Please conduct a thorough self-review before opening the PR.

- [x] I am confident that the code works.
- [x] I have updated documentation where appropriate.

## Summary

Added tx_ref to the ingress-egress-tracker

#### Non-Breaking changes

### CODE REVIEW

File: api/bin/chainflip-ingress-egress-tracker/src/witnessing/state_chain.rs [hunk 1]
```diff
@@ -14,6 +17,14 @@ use pallet_cf_ingress_egress::DepositWitness;
 use serde::Serialize;
 use utilities::rpc::NumberOrHex;
 
+#[derive(Serialize)]
+#[serde(untagged)]
+enum TransactionRef {
+	Bitcoin { hash: String },
+	Ethereum { hash: H256 },
+	Polkadot { transaction_id: PolkadotTransactionId },
```
Comment (niklasnatter) [inline api/bin/chainflip-ingress-egress-tracker/src/witnessing/state_chain.rs:25]: how would this get serialized?

**Review by niklasnatter — APPROVED** [review]

**Comment by marcellorigotti** [comment]
We decided to make some changes to the way the tx_ref is reported back to the chain, will update this PR once the other one is up and merged (it is just an implementation detail nothing changes for you @niklasnatter )

**Comment by marcellorigotti** [comment]
PR is updated and ready, with the latest changes made to transaction_succeeded extrinsic.

File: api/bin/chainflip-ingress-egress-tracker/src/witnessing/state_chain.rs [hunk 2]
```diff
@@ -14,6 +16,14 @@ use pallet_cf_ingress_egress::DepositWitness;
 use serde::Serialize;
 use utilities::rpc::NumberOrHex;
 
+#[derive(Serialize)]
+#[serde(untagged)]
+enum TransactionRef {
+	Bitcoin { hash: String },
```
Comment (dandanlen) [inline api/bin/chainflip-ingress-egress-tracker/src/witnessing/state_chain.rs:22]: @marcellorigotti 
Explicit conversion to String is a bit of an anti-pattern. I think a better solution here is to use `BitcoinTransactionHash` and define `Serialize` on that struct. The *serialization* should reverse the byte order. Then we can use the same struct for the tx_ref and tx_out_id. 

IMO this is less confusing than storing two different tx references.

**Review by dandanlen — APPROVED** [review]

**Pull request merged**